### ISSUE TITLE
Mark up text for automated string extraction

### ISSUE THREAD

**Issue by emmajclegg** [IATI/iatipublisher#1477]
This is a first step to translating IATI Publisher's interface into French and Spanish, following the approach discussed here: [#1420 ](https://github.com/younginnovations/iatipublisher/discussions/1420)

YI will prepare text for automated extraction from IATI Publisher, ODS will review and get it translated, then YI will reintegrate text back into IATI Publisher. 

# Tasks
- [x] Understanding the requirement for automated string extraction
- [x] #1560
- [x] #1562
- [x] Create UI for updating the system language
- [x] Add placeholders in public pages and add english texts from those pages in php files with proper folder structure
- [x] Write a script to read the language files and download xls file in 3 columns for English, French, Spanish
- [x] Upload the translated files in the language files by reading the XLS file
- [x] Test the public pages for translation
- [x] Add placeholders in remaining pages, common buttons and notifications and add english texts from those pages in a json file
- [x] Upload the translated files in the language files by reading the XLS file
- [x] Complete the translation of the system
- [x] Document how to add translation for future development and changes

# Test extraction by modules
- [x] #1638
- [x] Elements PLUS Formbuilder
- [x] Onboarding (completed in activity module)
- [x] #1642
- [x] #1639
- [x] #1641
- [x] Import csv/xml(wip)
- [x] Import xls (wip)
- [x] Export
- [x] #1640
- [x] #1647
- [x] #1650
- [x] #1653
- [x] #1664
- [x] #1658
- [x] #1659

**Comment by robredpath** [comment]
Following our conversation this week, I wanted to share an outline of the process that you'll need to follow. 

It's important that this is an automated process that is part of your standard workflow, so that any changes to text are detected and translated quickly in the future. 

In my experience, this is usually achieved by marking up the text in some way. See, for example, [this Django template](https://github.com/open-contracting/cove-ocds/blob/main/cove_ocds/templates/cove_ocds/base.html#L136-L140) from one of our projects, which wraps each sentence in `{% blocktrans %}` tags which signal to Django's i18n module that the string should be included in the translation process. Our workflow uses [.pot/.po files](https://www.drupal.org/community/contributor-guide/reference-information/localize-drupal-org/working-with-offline/po-and-pot-files).

I can see that there's already some [files with lists of strings](https://github.com/younginnovations/iatipublisher/blob/main/lang/en/validation.php) that look like a translation mechanism, which may be how this is already starting to be implemented? My instinct is that this is potentially quite a fragile and high-effort way to work, but it's up to you! 

Either way, once the system is in place, then each time we make an update, the process is:
* Extract strings
* Package strings that have changed in a .pot file
* Send to translators
* Wait
* Receive translated strings as a .po file
* Re-integrate strings into the software 

Apart from a few manual steps to authorise the translation and review the software output to make sure that nothing went wrong, this is an entirely manual process. 

I don't think that IATI Publisher necessarily has to have a .pot/po file - based process, but if you were to build one then it would be very close to what we're going to need once we work out the file details with the translation company. 

Does that make sense? I'm very happy to provide any more detail if it's useful.

**Comment by emmajclegg** [comment]
Thanks @robredpath - I assume you meant "Apart from a few manual steps to authorise the translation..., this is an entirely automated process" ? Otherwise, no questions from me

**Comment by Sanilblank** [comment]
Hello @robredpath 
Based on the template you have provided, it seems that you are trying to mention the localization mechanism of Django framework. For the translation process in IATI Publisher, when we initially started the process, we used a similar approach. Laravel, the framework used for the backend development, also provides a similar approach for localization and if the templating engine of laravel had been used for the frontend part as well, the template would look very much similar to what you have given as a template. However, since we are using Vue js, a little bit more complexity is added for displaying the text in the frontend side even though a similar templating structure is used by Vue as well.
Similar to how you have mentioned the use of .pot/.po files, we (in laravel) also use similar files that contain array of data for saving the translations in different files as required. The process and mechanism for both is quite similar, so I don't think we will need to go through the process of incorporating .pot/.po files directly into the system as it will require a lot more research on how it can be achieved. The translations will be stored in files within the system but since providing those files directly to you for translation could cause confusion about how to process them as they fall more into the technical category, we were thinking of writing a script that would take all the strings and place them in an excel file which would be provided to you and then you would add the translated strings and provide the file back to us, and finally a script to take those translations and put them in the format required by the system.
If you will required .pot/.po files for us to send the strings to be translated, we will need a bit of time to research how the files can be created, and then the rest of the process will be similar i.e. a script will generate the file which will be sent to you, you will add the translations and send the file back to us, and a script will take the translations and put them into the system.
We are a bit confused on the use of the word 'automated' in your title and description.
By automate I am assuming you mean that a file will contain all the english strings present in the system and if any changes occurs in the file, a script will detect the change and then generate the required excel or .pot/.po file which will be sent to you for translation. If my understanding correct? If it is not, could you provide a bit more explanation for this part.
Hope I have made things clear here, if there is something that seems a bit confusing, I would be glad to go a bit deeper in the explanation.

cc. @praweshsth @PG-Momik

**Comment by emmajclegg** [comment]
Thanks for the information here @Sanilblank . @robredpath is away until Aug 29th unfortunately, but I will see if anyone else in our team can help with the file format question in the meantime.

> We are a bit confused on the use of the word 'automated' in your title and description.
> By automate I am assuming you mean that a file will contain all the english strings present in the system and if any changes occurs in the file, a script will detect the change and then generate the required excel or .pot/.po file which will be sent to you for translation. If my understanding correct?

Yes, that's correct to my understanding. We remain in control of how often and when we run the re-translation, but your system should be capable of detecting what English text has and hasn't changed since the last translation.  

By the extraction and re-integration of text into IATI Publisher being done in an automated way, we mean via a script as opposed to any manual copy and pasting.

**Comment by Sanilblank** [comment]
@emmajclegg Thanks for the clarification.
I have another question regarding the extraction part. The system will generate the excel or another format file containing the english strings to be extracted which will be sent yo you. We will need a way to inform you about which strings have been added/changed since the last translation was done. So, we could do this in many ways, the first being generating only the strings which have been either added or updated (which will require translation) in the file, another way could be generating all the string in the file along with previous translations as well which will allow you to update the translations even for the ones which were already done previously. The second approach would give you more flexibility but it may be more difficult for you to see which texts actually require translating.
If you have any other ideas regarding this subject, we are open to hear them. Please have a look and confirm the process which we should move forward with.

cc. @praweshsth @PG-Momik

**Comment by emmajclegg** [comment]
Hi @Sanilblank - I don't want to give wrong information on this so will check with @robredpath once he's back (Aug 29th) and update here.

**Comment by robredpath** [comment]
Hi @Sanilblank! Thanks for this - it's really useful to understand what you're thinking. 

The exact format of the files doesn't really matter too much for us - I suggested .pot/.po files as they're fairly standard in our other applications and are straightforward to work with, but an .xlsx file would also be fine. The main thing is that the process is automated and repeatable. 

By "automated" what I mean is that we expect the list of strings to be generated directly from the source code by software, without any manual steps - and for the translated strings to similarly be re-integrated automatically. This means that the process is easily repeatable, so that a small update can be made easily and large updates aren't too much of a problem. 

We don't expect the automation to require zero human contact, but we want to make sure that everything gets translated as part of the regular updating process for the software: every time a form or button changes, or we add some new explanatory text, it should be translated promptly. 

By way of example, for our documentation platform we run one command to generate the .pot files that we send to the translators, and then we check in the translated files to git and re-run the build process to generate the multi-lingual website. This gives us a very high level of repeatability and consistency, and it's easy for us to do which encourages us to do it often - even for very small changes. 

In our documentation work we send the whole documentation site each time, and the translation platform figures out what's changed, and gets that translated. We then re-import the whole translated file back in. Our experience is that it's easier that way, rather than trying to manage lists of things that have changed. Ultimately, it is up to you, but that's our experience and recommendation. 

Hope that helps - do let me know if you have any further questions

**Comment by Sanilblank** [comment]
Hi @robredpath 
I think I understand what you are trying to say and I feel that we are on the same page regarding the automation process. As mentioned previously, we will be writing a script which will be responsible for checking the translations maintained in the system and will be able to generate an excel file consisting of all texts either translated or requiring translation which will be sent to you. You will perform the translations as required and will send the file back to us and we will use a script to simply take the translations and insert them into the system.

cc. @praweshsth @PG-Momik

**Comment by Sanilblank** [comment]
Hi @robredpath 
The above parts discussed between us are very clear now. For the part where we send the data present in the backend to the frontend, we researched online and found two methods.
1. The entire data is loaded in the app.blade.php file and saved as global data. Then, the FE uses the data for showing the texts throughout the system. This increase the load in each page, so we could put this in cache and then use the data everywhere. Still, the data stored in cache would be very high and so this process may not be feasible. Also, when the language is changed by the user, the cache data is delete.
2. The BE will have apis for sending the translated texts to the FE. When a page loads, the apis for the required translated texts will be called. When an api for a certain set of translations is called, they will be store on the backend cache in redis so that next time no processing part is required. When a user changes the language, the cache data will be deleted. This process will not send all the translated texts data to the FE immediately which will help to reduce the load. We are leaning towards using this method.

This message is just to update you regarding the findings we have had and to give you an update about how we are proceeding for this feature.

cc. @praweshsth @PG-Momik

**Comment by emmajclegg** [comment]
@Sanilblank - can you update on how this automated translation work is going please? We'd committed to implementing French & Spanish translation by the end of this year so I'm worried about delivery timelines slipping. 

I'm keen to review the extracted English text (the step before getting it translated) as soon as possible, so I discussed briefly with Bibha & Momik yesterday the possibility of you sharing the text you've extracted so far, rather than waiting until all user-facing text has been incorporated. This would mean I could start reviewing earlier.

Any blockers or questions, let me know. 

I'm also trying to close old IATI Publisher issues that are no longer relevant - let me know if it's ok to close #885 and #1279? These were the previous issues related to translation that I assume have been replaced by this current issue.

**Comment by Sanilblank** [comment]
@emmajclegg Hello Emma
I have been informed about performing the translation in sections instead of performing everything at the same time.
I will be extracting the translation texts for the public pages and will send them to you by Wednesday at the latest. Once you confirm the format for the excel and the flow for the public pages is good, the remaining parts will be completed.

> I'm also trying to close old IATI Publisher issues that are no longer relevant - let me know if it's ok to close https://github.com/younginnovations/iatipublisher/issues/885 and https://github.com/younginnovations/iatipublisher/issues/1279? These were the previous issues related to translation that I assume have been replaced by this current issue.

Yes those issues can be closed.

cc. @PG-Momik @BibhaT

**Comment by Sanilblank** [comment]
Hello @emmajclegg ,
As Momik must have mentioned yesterday, I was on leave for some days because of health issues and did not have a chance to properly work on the translation tasks.
From the recent comment it is clear what changes need to be implemented based on the sample of the public pages translation which was sent previously.
The process of performing the translation now is mostly a manual task and requires the copying of the texts in different sections, pasting in the language files and replacing the original files with the placeholders. Since most of the remaining task is manual work, it is difficult to give an exact estimation about how long it will take to complete the entire system translation.
Since you have mentioned that this task takes priority, we will be giving more time and will be working towards finishing the tasks in a quick and effective manner. Also, if the estimations are required, we could finish a module like settings or organizations and then move towards creating an estimation based on the time taken to complete that one particular module.
Thank you,
Sanil Manandhar

cc. @PG-Momik @BibhaT

**Comment by emmajclegg** [comment]
Thanks @Sanilblank - I appreciate you've been off recently, that's no problem.

I don't need an exact estimation for this at this point - the main thing is I'd like us to be making visible progress on it, rather than letting it potentially drag on for months.

Just let me know when you've picked the module to work on first and when roughly I should be expecting to receive a text file to review (as it helps me plan). I expect we'll want to test run the entire extraction, translation, reintegration process on a single module first which, as you say, will help us all understand time and effort required for the remaining ones.

**Comment by emmajclegg** [comment]
To summarise where I think we got to on the questions from today's call:

- **IATI Standard element names** - if these appear as part of a sentence in IATI Publisher, we will extract and translate the whole sentence (I think this makes sense for users' understanding). If the element names are standalone in the interface, we will still extract the strings but are unlikely to translate these into French and Spanish in the short term. We may translate them eventually, so want to keep this as an option.

- **IATI codelists** - ODS are likely to translate IATI-maintained codelists into French & Spanish, but not externally maintained ones (e.g OECD DAC lists). If and when we do get the codelists translated, we want to make sure IATI Publisher can display these. We worked on syncing codelists in issue #1407 , so I believe IATI Publisher does detect automatically when IATI Standard codelists change ?

Any other questions, or anything to add, just let us know @PG-Momik @BibhaT 

cc' @robredpath

**Comment by PG-Momik** [comment]
@emmajclegg  Adding to this. On the call I mentioned that I wasn't sure if other codelist were besides OrganizationRegistrationAgency being sync'd. I've confirmed that the  codelist are being synced as well. :+1:

**Comment by PG-Momik** [comment]
@emmajclegg A sheet has been shared with the current extracted contents for the completed modules. Please have a look.

cc: @BibhaT

**Comment by emmajclegg** [comment]
Ok thanks a lot @PG-Momik - I'll have a look over the next few days, prioritising the sheets labelled green in the [extracted text spreadsheet](https://docs.google.com/spreadsheets/d/1PZ4bPt8Bxt5_rm8Xx7xJ6z6dj-xdc4WD0hEUNG8gnlo/edit?gid=859630918#gid=859630918).

I'll highlight yellow and add a comment to cells that I either don't understand or that I think are causing duplication. For example, I think it's important that standalone IATI-specific strings like "IATI organisation identifier", "publisher ID" and "default language" are translated only once and applied consistently across the tool.

cc'ing @robredpath for info. Rob - let me know what you think we need to run a first test of the translation and reintegration loop. "adminHeader" is the simplest sheet in that extracted text workbook, if useful as an easy example. Otherwise I'll let you know when there's a few sheets de-duplicated where I've reviewed the English text.

**Comment by emmajclegg** [comment]
@PG-Momik - to update, I've looked over the remaining green (i.e. nearly finished) sheets in the [extracted text spreadsheet](https://docs.google.com/spreadsheets/d/1PZ4bPt8Bxt5_rm8Xx7xJ6z6dj-xdc4WD0hEUNG8gnlo/edit?gid=859630918#gid=859630918) and have left a few more comments.

I haven't edited any of the English text yet as it sounds like it make sense for YI to re-extract an updated version of the spreadsheet before I do that (to save me re-doing the review before @robredpath sends the text for translation).

Happy to discuss any questions tomorrow.

**Comment by emmajclegg** [comment]
@PG-Momik - thanks for sharing the latest spreadsheet of extracted text ([Extracted Sheet - Jan 17](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?gid=570899961#gid=570899961)). I assume this was for me to have a look.

Again, I've left a few comments to check where certain text appears in the interface and flag a few areas where it could be simplified.

- Cells highlighted orange - I'm questioning whether these contain user-facing text, as they look more like log/error messages.
- Cell highlighted yellow - suspected duplication (I will have left a comment in the sheet)

Happy to discuss more on Wednesday (I'm not around tomorrow, Tues 21st)

**Comment by emmajclegg** [comment]
@PG-Momik @BibhaT - I'm summarising points from today's call below - anything that doesn't look right, let me know. (cc' @robredpath )

**Restricting translation to user-facing text**:
- It is not always clear to the developers marking up text for extraction whether it is user facing or not. YI would need extra time to check in some cases
- I advised that we remove text from [Extracted Sheet - Jan 17](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?gid=570899961#gid=570899961) that we're 100% sure is not user-facing in the IATI Publisher UI (e.g. API messages). I think it's fine for YI to spend a few extra hours checking cases that aren't clear (as we want to minimise the text we need to translate), but it's not worth spending days on it.
- To avoid unnecessary translation, it makes sense to remove generic Laravel messages on the [General sheet](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?gid=2002881026#gid=2002881026) from extraction if they are not actively being used in IATI Publisher

**Simplifying common messages that contain element names:**
- We discussed different approaches to simplifying cases where the same text appears often, just with a different IATI element name in it - e.g. cells C28-C148 of the ["activity_detail"](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?gid=263168710#gid=263168710) sheet.
- In these cases, we can either 1) remove the element name from the text to make it generic, or 2) include the element name in the text as a :var variable
- Approach 1 would be sufficient for the data entry form messages on the [activity_detail](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?disco=AAABb3u58Zw) sheet. 
- Validation messages on the [General](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?gid=2002881026#gid=2002881026) sheet, however, are more specific to IATI elements and attributes. Approach 2 (introducing variables) may help reduce the amount of text here, but it is unclear whether this is worth the effort. As a minimum, I was planning to make sure the validation messages are user-friendly by removing @ symbols from element/attribute names to make the text more readable to users.

**Testing the translation and text re-integration process**:
- We would like to test the translation and re-integration process soon, to identify any problems and get feedback from the translators
- The easiest modules to start with would be "public", "adminHeader" and "footer"
- YI will let ODS know when these modules are ready (i.e. unlikely to change further in the spreadsheet), then I will edit the English text where needed (in a copy of the spreadsheet, and without adding/deleting rows), and share with Rob for translation.

**Comment by emmajclegg** [comment]
@PG-Momik - to update, it's looking increasingly likely that we'll remove or significantly reduce the text on IATI Publisher's public facing pages, based on conversation in #1543, so I wouldn't spend a lot of time tidying up the ["Public" sheet](https://docs.google.com/spreadsheets/d/1CQfZkhaYn8fnMMpameCuLkk2HusbFWqUwl0jEFHhFkY/edit?gid=1981972340#gid=1981972340) text in the meantime.

We can confirm next week and discuss if other sheets in the Extracted Text spreadsheet would be more sensible to use for initial testing of the translation process.

**Comment by PG-Momik** [comment]
Hello @emma, As mentioned on the previous call, sheet: Header, Footer and Public wont need a lot of tyding up and are okay for review +  translation,  since they are most likely to not change content-wise. I don't think changes in #1543  affects the content of the public sheet.  We can discuss it on the next call.

**Comment by emmajclegg** [comment]
@PG-Momik - thanks for sharing the updated [Extracted Sheet - Feb 10](https://docs.google.com/spreadsheets/d/1ys37g_TWr0pAewE1CKI6iWPcElHEL67nivCgF1bDaiE/edit?gid=722227114#gid=722227114)

I've taken [a copy](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114) and will review the English text, only making edits to the "EN" column. I'll use this as a chance to make text more concise, check grammar and ensure consistency in the way things are described.

I'll aim to finish this review tomorrow and will let you know when done. 

I suggest, to break the translation testing process down into smaller steps, that we reintegrate my changes to English text into IATI Publisher first and check that works ok. Then we can move on to getting feedback from the translators.

**Comment by emmajclegg** [comment]
Hi @PG-Momik - I’ve finished reviewing the English text here: [(ODS reviewed) Extracted Sheet Feb 10.xlsx](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114)

I’ve only made edits to the "en" columns, and have not added or removed any rows, so hopefully this is ok to reintegrate into IATI Publisher. 

To summarise, I have:
- Reviewed all text for accuracy and grammar. If I don't understand the text, I've left it unchanged and tagged you in the comment Momik.
- Standardised placeholders for dropdown menus to “Select an option”. This reduces text we’re translating and adds more consistency for users
- Removed some placeholders or help text that seem duplicative of text elsewhere, e.g. for default values. (This is without deleting any rows in the spreadsheet)
- Made sure help text is consistent when the same fields appear in IATI Publisher's 'Get Started' onboarding guidance and the 'Settings' pages
- Rewritten some text related to XLS imports, as this was not the easiest to understand before. Note, this includes simplifying the XLS template names to "basic activity elements", "results", "indicators" and "periods"
- Reduced variation in some of the validation messages on the "general" sheet, though this requires more work (I've tagged you in a few questions)

Any cells coloured yellow are to highlight where i’ve standardised (i.e. introduced duplicate text) or removed text. 

Any cells coloured grey are for text that I don't think we need to translate - the text is either being removed from IATI Publisher soon or I'm very confident it is not end user-facing.

Given that I've updated quite a lot, I recommend we reintegrate this updated text into IATI Publisher, before doing a final tidy up of the spreadsheet. I assuming this re-integration of text is fairly automated now?

Happy to discuss or answer questions on tomorrow's call.

**Comment by emmajclegg** [comment]
@PG-Momik - to update, I've gone through the "general" sheet in the [(ODS reviewed) Extracted text](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114) and have standardised the validation messages as best as I can.

Anything yellow on that sheet should be highlighting a duplicate. If we can remove the worst of the duplication for the final version of the spreadsheet great, though it doesn't have to be perfect.

I won't make any further changes to the English text now, so go ahead and reintegrate back into Publisher when ready.

**Comment by PG-Momik** [comment]
@emmajclegg  Extracted text on Feb 25. Leaving a comment here since I cannot comment on the sheet for some reason.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik . The format of [(ODS reviewed) Extracted text](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114) looks good to me now. I've made a few edits to the new pieces of text added since the last extraction.

@robredpath - this workbook is now in a good state for you to have a look at ^^ 
I'm happy to catch up before you share it with the translators, to answer any questions and tackle how we're going to deal with new text appearing in IATI Publisher.

@PG-Momik - as a next step, can we prepare the updated English text for release in IATI Publisher? This will be beneficial to end users straight away, so I'd like to release it soon rather than wait until the French and Spanish is ready.

**Comment by PG-Momik** [comment]
@emmajclegg This has been deployed to staging.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik - great to see this. I've run out of time today but will have a proper look tomorrow and feedback.

**Comment by emmajclegg** [comment]
@PG-Momik - I've focused on the English pages (without going through every single one), and noticed the things below.

Correct what you need to in the interface and extracted text, then I'm happy to do a final look over the [(ODS reviewed) Extracted text](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114) sheet.

1. "My organisation is new to IATI" registration route >> the following help text is missing / misplaced:

- [ ] "Organisation Registration Agency" (step 1)
- [ ] "Contact Email" (step 2)
- [ ] "Default Language" (step 3) - probably doesn't need help text, just unsure why the "?" icon is there

![Image](https://github.com/user-attachments/assets/ce4f2db1-2ef3-46cb-8e17-713801cb032b)

2. Similar to above, the "My organisation is already registered with IATI" also has some missing help text:

- [ ] "Publisher ID" (step 1)
- [ ] "Country" (step 1)
- [ ] "Default Language" (step 2) - as above, the "?" icon can probably just be removed.

![Image](https://github.com/user-attachments/assets/cf8c250a-5da5-4422-bea9-053a0c3eef47)

3. "Get Started" guidance

- [ ] HTML link is visible for "Settings" at the bottom of Step 2. There's no link for the "Organisation Data" page in step 3, so we can probably just remove the HTML here.

![Image](https://github.com/user-attachments/assets/08883ed3-cef2-4a95-a5ec-10a9bc40d4d0)

- [ ] "Default Hierarchy" and "Default Finance Type" - help text under the data entry fields is not needed (the info's already included in the "?" icon text)

![Image](https://github.com/user-attachments/assets/4092a2d1-052f-42bc-b6fc-6baea6a53e11)

- [ ] "Humanitatian" field is repeated - this should be "Default Flow Type"

![Image](https://github.com/user-attachments/assets/a7d06c9f-7662-4214-93eb-0cddc4172d0d)

- [ ] "Default Aid Type" has wrong help text
- [ ] "default-tied-status" is written as an element name, whereas the other field labels aren't. Ideally we'd write labels on this page in the format "Default Tied Status", as it appears now on the production site

![Image](https://github.com/user-attachments/assets/eccd217d-6fed-41d3-aa6e-a93f35962df7)

- [ ] Step 3 - "reference" appears twice, when it should be "Type"

![Image](https://github.com/user-attachments/assets/26108c11-003b-460a-8c13-ec5d6ac30bde)

4. Settings pages

- [ ] "Publisher ID" is missing help text

![Image](https://github.com/user-attachments/assets/567edb25-6f4a-45bc-8acf-1e7a5abcba1d)

- [ ] "Default Hierarchy" help text underneath the data entry field can be removed
- [ ] "Default Flow Type" label is missing
- [ ] "linked-data-uri" appears with hyphens as an element name - same as above, ideally all field labels on this page would appear plain English, capitalised, like on the production site ("Linked Data URI")

![Image](https://github.com/user-attachments/assets/e67ad51a-79dd-4378-8757-aa503bd13898)

5. Add a new activity manually
- [ ] "title" and "narrative" are missing help text when clicking on "?"
- [ ] "Activity Identifiers" shouldn't be plural >> "Activity Identifier"
- [ ] "Iati-Identifier" looks odd capitalised - this should be "IATI Identifier" for consistency with other labels on the page

![Image](https://github.com/user-attachments/assets/72755cef-f941-46c7-877e-dff43f6a0e31)

We're getting there! Any questions, just shout.

**Comment by PG-Momik** [comment]
@emmajclegg This has been redeployed to staging.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik - I can see my previous points have been fixed. A few more:

- [ ] "My organisation is new to IATI" registration route - Step 3 has wrong label ("Email Verification")

![Image](https://github.com/user-attachments/assets/32ecb561-442f-43bb-a510-b39968e45aed)

Some cases of what look like missing FR_ / ES_ placeholders:

- [ ] Publishing Settings: "IATI Registry Information", "Pending / Correct / Incorrect"
(I don't know how the "Support" widget works - is the word "Support" translatable or is it linked to Zendesk?)
![Image](https://github.com/user-attachments/assets/9ff352ff-cc9c-473b-ab94-d8280bafe49c)

- [ ] Import activities from .XLS: "Period" should be plural ("Periods"), and FR placeholder is missing
![Image](https://github.com/user-attachments/assets/66681373-0bc5-4d27-b499-cf38186aa48a)

I've found a few cases where text reused on the Production site needs separately into separate messages:

- [ ] Currently, the CSV and XLS template download '?' help text is the same.

What I've updated it to makes sense for XLS... 

![Image](https://github.com/user-attachments/assets/8e24dc88-a508-4a01-a940-3097d180ae27)

...but not for CSV
![Image](https://github.com/user-attachments/assets/594ad877-472f-4770-9c54-a9ea2f1ca072)

Can we have a different message for the CSV help text - e.g.
_"This template covers basic activity data and transactions. All possible data elements of the IATI Standard are included for flexibility, but it is not necessary to populate all columns.  Do not make any changes to the structure of the template when you enter your data."_

- [ ] Override default values within an activity: similarly, the three pieces of help text that I've highlighted below are reused from the Settings page (I think). They're a bit confusing currently as they talk about applying values to every activity. 

![Image](https://github.com/user-attachments/assets/65438477-34b5-4501-b60c-4058b2b409f1)

I suggest the following text instead, which would only apply to this override page:

Hierarchy: 
_If you are reporting both programmes (parent activity = 1) and projects (child activity = 2), enter the hierarchical level of this activity._

Budget not provided:
_If you are not able to provide budgets for this activity, select the reason here._

Humanitarian:
_Select "Yes" if this activity wholly or partially addresses a humanitarian crisis._

**Comment by PG-Momik** [comment]
@emmajclegg This has been redeployed to staging with the added changes.

**Comment by emmajclegg** [comment]
Great, thanks @PG-Momik - all looking good!

Only thing I've noticed is a missing FR_ / ES_ placeholder for the periods template:
(I'm aware we're not translating the XLS templates themselves)

![Image](https://github.com/user-attachments/assets/e83f33e9-fc55-4d45-95de-23a7d597c7eb)

Can you share the latest version of the extracted text spreadsheet when ready? I'll do a final review for typos and accuracy, now that it's clear where the text is appearing in the interface.

**Comment by PG-Momik** [comment]
@emmajclegg, this has been redeployed. I've updated the extracted sheets as well.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik .

I've reviewed and made some edits to the [(ODS reviewed) Extracted text](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114) (particularly as some of my text edits last week hadn't been picked up). None of my changes require changes to the spreadsheet structure or position of text in the interface, so hopefully simple to just reintegrate.

Next steps:
- Reintegrate the extracted text spreadsheet above
- Remove the language switcher links (FR/ES) in the header to prepare the updated English interface for release
- Replace links to the public facing pages (see new issue: [#1772](https://github.com/IATI/iatipublisher/issues/1772)).

**Comment by Sanam-07** [comment]
Hi @emmajclegg 

Could you list out some of the sheet name and cell no. that had not been picked up. This will help me verify if those text have been picked up in the current deployment. 

Thank you.

**Comment by emmajclegg** [comment]
Hi @Sanam-07 

In the [(ODS reviewed) Extracted text](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=722227114#gid=722227114) sheet, some examples of text cells not previously reintegrated are:

- "Common" sheet, cells C323-324 (with the email new joiner messages)
- "Common" sheet, cells C333, 336, 338-340 (indicator, period, recipient country/region messages)

There weren't many cases, so I'm not assuming it's a problem with your reintegration process. There were a few days last week where @PG-Momik was adding extra text cells to the workbook and I was reviewing overnight. So I suspect it's just some of those edits which missed reintegration.

**Comment by emmajclegg** [comment]
For info @Sanam-07 - I spotted the text strings below when changing an organisation-identifier on the staging site today:

![Image](https://github.com/user-attachments/assets/5ba0c12a-4332-4c6d-b512-771970ff5889)

**Comment by emmajclegg** [comment]
Thanks @Sanam-07 - all the reintegrated text looks ok to me on the Staging site now (I've checked the examples I gave in the [comment above](https://github.com/IATI/iatipublisher/issues/1477#issuecomment-2725603306)).

I can also see that the French/Spanish language switcher links have been removed and a link to the new help docs added on the public-facing homepage, which is great.

This updated English interface looks ready to release at this point, but let me know if there's anything specific you're still checking.

**Comment by emmajclegg** [comment]
@Sanam-07 - I spoke too soon in my last comment!

This is my view on the staging site after having imported an XLS file:

![Image](https://github.com/user-attachments/assets/08b59228-efe6-4477-b758-716abe4fad65)

Each of the three pieces of text I've highlighted in red should say "activities" as far as I can see. 

I also don't know what's happened to the activity titles in the list (the XLS file I uploaded is attached to my last comment in https://iati.zendesk.com/agent/tickets/46622)

**Comment by Sanam-07** [comment]
Hi @emmajclegg ,

The issue has been resolved, and the fix has been deployed to the staging. Please check and let us know if everything looks good on your end.

Thanks!

**Comment by emmajclegg** [comment]
Welcome back @PG-Momik. 

On this issue - I don't know if it's related but fyi that the Transaction List page selection bar appears missing on the staging site:

![Image](https://github.com/user-attachments/assets/e8013c69-fd74-49e0-9004-05362c597d3a)

It looks fine on production.

**Comment by PG-Momik** [comment]
Hi @emmajclegg . This has been fixed and deployed to staging. I've also updated the review sheet as well. Let us know if there is any inconsistencies remaining in either publisher or the excel sheet.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik - I've had a quick look over the extracted text sheet and the staging site and all looks ok. I can see the Transaction List page issue is fixed.

One thing I was going to mention on the call tomorrow, which may be easiest addressed as part of this issue, is that I've noticed some end-user confusion about .xlsx vs. .xls for IATI Publisher's import templates (the files downloaded are in .xlsx format, but the IATI Publisher interface talks about .xls). Some users are deliberately saving their Excel templates in the older file format, which I don't think is necessary and may be losing some of the formatting & dropdown menus for them.

Provided there's no reason for this discrepency, can we update user-facing .xls references to .xlsx in the interface? I've already done this in the [extracted text sheet](https://docs.google.com/spreadsheets/d/1TNJ9jmNx92TtIPcozVx0RbLJSc1ulmkwUXS1kaewGP0/edit?gid=1526099468#gid=1526099468) if easiest to just reintegrate.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik - all looks good to release!
